core/trino-main/src/main/java/io/trino/security/ForDefaultSystemAccessControl.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.security;
+
+import javax.inject.Qualifier;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target({FIELD, PARAMETER, METHOD})
+@Qualifier
+public @interface ForDefaultSystemAccessControl
```

Comment:
I am not sold on the name. Maybe TheDefaultSystemAccessControl? or SystemAccessControlDefault
Or maybe DefaultSystemAccessControl if we rename the `DefaultSystemAccessControl` class?

Reply:
I think the name is fine. See `AccessControlManager` constructor parameter:
```
            @ForDefaultSystemAccessControl SystemAccessControl defaultAccessControl)
```
Where it says that this `SystemAccessControl` has to be a default system access control.

Reply:
i saw that, whatever

Reply:
BTW wouldn't it make sense to inject SAC default Factory, instead of an actual implementation?

Reply:
No. Then it would require to inject the configuration which is actually not needed.

Reply:
the configuration is empty, but the gain would be the laziness - why would you want to instantiate the SAC instance that you're not going to use.

Reply:
Good point. However, so far laziness is not needed. I am not sure if it make sense to introduce it now.

Reply:
I think it would help code clarity, especially since we have explicit SAC Factory type, so until now we were only instantiating SACs that are actually used.

Reply:
https://github.com/trinodb/trino/pull/9071